api/src/probe-routes.ts

```diff
@@ -1,8 +1,70 @@
+import fs from "node:fs";
+
+import { getIgnoredPaths, shouldIgnoreFile } from "./lib/utils.js";
+
+let debounceTimeout: NodeJS.Timeout | null = null;
+
+// biome-ignore lint/suspicious/noExplicitAny: Trust me, this is easier
+function debounce<T extends (...args: any[]) => void>(func: T, wait: number) {
+  // biome-ignore lint/suspicious/noExplicitAny: Trust me, this is easier
+  return (...args: any[]) => {
+    if (debounceTimeout) {
+      clearTimeout(debounceTimeout);
+    }
+    debounceTimeout = setTimeout(() => {
+      func(...args);
+    }, wait);
+  };
+}
+
+/**
+ * Since we are calling the route probe inside a file watcher, we should implement
+ * debouncing to avoid spamming the service with requests.
+ *
+ * HACK - Since we are monitoring ts files, we need a short delay to let the code
+ *        for the service recompile.
+ */
+const debouncedProbeRoutesWithExponentialBackoff = debounce(
```

Comment:
debounced this at first because i realized we were watching `.wrangler` which caused many many probes to go off. after implementing minimatch logic that mitigated the need for this but i kept it anyhow.